server/pulp/server/async/tasks.py

```diff
@@ -255,7 +255,7 @@ def apply_async_with_reservation(self, resource_type, resource_id, *args, **kwar
         # To avoid the race condition where __call__ method below is called before
         # this change is propagated to all db nodes, using an 'upsert' here and setting
         # the task state to 'waiting' only on an insert.
-        task_status.save(fields_to_set_on_insert=['state', 'start_time'])
+        task_status.save(fields_to_set_on_insert=['state', 'start_time'], safe=True)
```

Comment:
We do actually almost work with replica sets, except for the Celery results backend bug. I suspect we'll have that bug fixed soon since it's high priority.

I agree that we should stop using the safe kwarg since it is deprecated and start using write concern.

Unfortunately, write concern set to 1 is not safe either so I recommend against the w:1 setting. This will only ensure that the write makes it to a single node and not to the other nodes. If there is a conflict, there is no guarantee that your write will "win", but your code will believe it did. With MongoDB 3.0, there is a w:majority setting, but we unfortunately cannot do that.

What we need is some code that calculates the majority for us that all of our Mongo queries can use. Something similar to floor(len(db_seeds)/2)+1. If we put something like that in pulp.server.db somewhere, then everyone can benefit. It could be calculated as a constant, or a function. Then we can set w to that value and be sure that we are getting our writes to a majority of nodes.